### PR TITLE
[Search Header] Change cartridge conf of advanced search dynamically

### PR DESCRIPTION
**Pull request by JohnnP**
## [Search Header] 

This modification allows to change the cartridge configuration of advanced search dynamically.
The compatibility with previous version is maintained.

### CODE REVIEW

**Comment by c3dr0x** [comment]
@JohnnP it should work, will test it.

Could you also correct the export (use `export default` instead of `module.exports`) ?

**Comment by Hartorn** [comment]
@JohnnP Please try to avoid typo in commit message :) 
@c3dr0x @JohnnP Just an idea : we could add an option, so that when the component unmount, we could go back the old cartridge, no ? Or it's not a use case ?

**Comment by JohnnP** [comment]
It's could be useful if the application has an unique header. But if each pages have their own header, the new cartridge can be different from the old.

File: src/page/mixin/cartridge-behaviour.js [hunk 1]
```diff
@@ -1,17 +1,15 @@
...
 
     /**
      * Updates the cartridge using the cartridgeConfiguration.
      */
-    _registerCartridge() {
-        this.cartridgeConfiguration = this.cartridgeConfiguration || this.props.cartridgeConfiguration;
-
-        if (!isFunction(this.cartridgeConfiguration)) {
-            this.cartridgeConfiguration = () => ({});
+    _registerCartridge(props = this.props) {
```
Comment (Hartorn) [inline src/page/mixin/cartridge-behaviour.js]: Smart one.

File: src/page/mixin/cartridge-behaviour.js [hunk 2]
```diff
@@ -25,14 +23,20 @@ module.exports = {
...
     },
 
     /**
      * Registers the cartridge upon mounting.
      */
     componentWillMount() {
         this._registerCartridge();
+    },
+
+    componentWillReceiveProps(nextProps) {
```
Comment (Hartorn) [inline src/page/mixin/cartridge-behaviour.js]: Please comment your functions next time (I think you can inherit doc from React for lifecycle, using @inheritdoc)

**Pull request merged**